src/java/org/apache/cassandra/db/SystemKeyspaceMigrator50.java

```diff
@@ -0,0 +1,124 @@
+
+    private SystemKeyspaceMigrator50()
+    {
+    }
+
+    public static void migrate()
+    {
+        migrateCompactionHistory();
+    }
+
+    @VisibleForTesting
+    static void migrateCompactionHistory()
+    {
+        migrateTable(false,
+                     SystemKeyspace.COMPACTION_HISTORY,
+                     SystemKeyspace.COMPACTION_HISTORY,
+                     new String[]{ "id",
+                                   "bytes_in",
+                                   "bytes_out",
+                                   "columnfamily_name",
+                                   "compacted_at",
+                                   "keyspace_name",
+                                   "rows_merged",
+                                   "compaction_properties" },
+                     row -> Collections.singletonList(new Object[]{ row.getTimeUUID("id") ,
+                                                                    row.has("bytes_in") ? row.getLong("bytes_in") : null,
+                                                                    row.has("bytes_out") ? row.getLong("bytes_out") : null,
+                                                                    row.has("columnfamily_name") ? row.getString("columnfamily_name") : null,
+                                                                    row.has("compacted_at") ? row.getTimestamp("compacted_at") : null,
+                                                                    row.has("keyspace_name") ? row.getString("keyspace_name") : null,
+                                                                    row.has("rows_merged") ? row.getMap("rows_merged", Int32Type.instance, LongType.instance) : null,
+                                                                    row.has("compaction_properties") ? row.getMap("compaction_properties", UTF8Type.instance, UTF8Type.instance) : ImmutableMap.of() })
+        );
+    }
+
+    /**
+     * Perform table migration by reading data from the old table, converting it, and adding to the new table.
+     * If oldName and newName are same, it means data in the table will be refreshed.
+     * 
+     * @param truncateIfExists truncate the existing table if it exists before migration; if it is disabled
+     *                         and the new table is not empty and oldName is not equal to newName, no migration is performed
+     * @param oldName          old table name
+     * @param newName          new table name
+     * @param columns          columns to fill in the new table in the same order as returned by the transformation
+     * @param transformation   transformation function which gets the row from the old table and returns a row for the new table
+     */
+    @VisibleForTesting
+    static void migrateTable(boolean truncateIfExists, String oldName, String newName, String[] columns, Function<UntypedResultSet.Row, Collection<Object[]>> transformation)
```

Comment:
@driftx does this method differ from the one in 40 migrator? What about having abstract migrator these two extends and put that method there so we can reuse?